File: R Programming/Rock_Paper_Scissors_Game.R
Rock_Scissors_Paper <- function(){
  
  message("Welcome to Rock Scissors Paper game!")  
  choice <- c("Rock", "Scissors", "Paper")
  Win <- 0
  Loss <- 0
  Draw <- 0
  
  while (TRUE) {
    
    user_hand <- readline("Choose your move: Rock, Scissors, Paper, Quit: ")
    computer_hand <- sample(choice, size = 1)
    
    if (user_hand == "Rock" | user_hand == "Scissors" | user_hand == "Paper") {
        cat("User hand:    ", user_hand, "\n")
        cat("Computer hand:", computer_hand, "\n")
        
        if (user_hand == computer_hand) {
          cat("Draw!")
          Draw <- Draw +1
        } else if (user_hand == "Rock" & computer_hand == "Scissors"){
          cat("You win!")
          Win <- Win + 1
        } else if (user_hand == "Rock" & computer_hand == "Paper"){
          cat("You loss!")
          Loss <- Loss + 1
        } else if (user_hand == "Scissors" & computer_hand == "Rock"){
          cat("You loss!")
          Loss <- Loss + 1
        } else if (user_hand == "Scissors" & computer_hand == "Paper"){
          cat("You win!")
          Win <- Win + 1
        } else if (user_hand == "Paper" & computer_hand == "Rock"){
          cat("You win!")
          Win <- Win + 1
        } else {
          cat("You loss!")
          Loss <- Loss + 1
        }
        
        message("\nWin:  ", Win)
        message("Loss: ", Loss)
        message("Draw: ", Draw)
    
        
    } else if (user_hand == "Quit") {
      message("Good bye, see you next time!")
      break
    } else {
      message("Please choose your move")
    }
      
  }
}

Rock_Scissors_Paper()

File: Python Programming/Rock_Paper_Scissors_Game.py
import random
user_hand = []

def rock_paper_scissors():

    name = input("What is your name? ")
    print(f"\n Hi! {name}. Welcome to rock paper scissors game.\n Please choose your move: rock, paper, scissors (type quit to exit the game) \n Let's Play!")

    draw = 0
    win = 0
    lose = 0

    while True:

        user_hand = input("\n Choose rock, paper, scissors: ")

        if user_hand == "rock" or user_hand == "paper" or user_hand == "scissors": 
            
            print(f"\n > user_hand: {user_hand}")
            computer_hand = random.choice(["rock", "paper", "scissors"])
            print(f" > computer_hand: {computer_hand}")

            if user_hand == computer_hand:
                print(" > Draw!")
                draw = draw + 1
            elif user_hand == "rock" and computer_hand == "paper":
                print(" > You lose!")
                lose = lose + 1
            elif user_hand == "rock" and computer_hand == "scissors":
                print(" > You win!")
                win = win + 1
            elif user_hand == "paper" and computer_hand == "rock":
                print(" > You win!")
                win = win + 1
            elif user_hand == "paper" and computer_hand == "scissors":
                print(" > You lose!")
                lose = lose + 1
            elif user_hand == "scissors" and computer_hand == "rock":
                print(" > You lose!")
                lose = lose + 1  
            else:
                print(" > You win!")
                win = win + 1

            print(f"\n win = {win}")
            print(f" lose = {lose}")
            print(f" draw = {draw}")

        elif user_hand == "quit":
            print(f"\n Goodbye {name} :D")
            print(f"\n win = {win}")
            print(f" lose = {lose}")
            print(f" draw = {draw}")
            break
        else:
            print("\n > Please try again")
